nixos/doc/manual/release-notes/rl-2009.xml

```diff
@@ -212,7 +212,20 @@ environment.systemPackages = [
        Be aware that backwards state migrations are not supported by Deluge.
      </para>
    </listitem>
-
+   <listitem>
+     <para>
+       Now nginx web server started in sandbox mode. By default, recording is
+       allowed in <literal>/var/spool/nginx</literal>. To allow writing to other folders,
```

Comment:
```suggestion
       Now nginx web server started in sandbox mode. By default, write access to
       <literal>/var/spool/nginx</literal> is allowed. To allow writing to other folders,
```

Reply:
Also, this is not `/var/spool/nginx`, but whatever `cfg.stateDir` is currently pointing to.

IIRC, we wanted to move this to `/var/lib/nginx` at some point of time.

Reply:
> IIRC, we wanted to move this to `/var/lib/nginx` at some point of time.

such a variant?
```
commit fbb86293947d326c4af08a214ecf41da40a80a36
Refs: [test-nginx], 18.09-beta-70407-gfbb86293947
Author:     Izorkin <izorkin@elven.pw>
AuthorDate: Sun Apr 19 17:50:11 2020 +0300
Commit:     Izorkin <izorkin@elven.pw>
CommitDate: Sun Apr 19 17:51:22 2020 +0300

    nginx: change logs path
---
 pkgs/servers/http/nginx/generic.nix                    | 12 +++++++++++-
 pkgs/servers/http/nginx/nix-skip-check-logs-path.patch | 27 +++++++++++++++++++++++++++
 2 files changed, 38 insertions(+), 1 deletion(-)

diff --git a/pkgs/servers/http/nginx/generic.nix b/pkgs/servers/http/nginx/generic.nix
index 67a914b6a98..63408be6a21 100644
--- a/pkgs/servers/http/nginx/generic.nix
+++ b/pkgs/servers/http/nginx/generic.nix
@@ -68,6 +68,14 @@ stdenv.mkDerivation {
     "--with-http_stub_status_module"
     "--with-threads"
     "--with-pcre-jit"
+    "--http-log-path=/var/log/nginx/access.log"
+    "--error-log-path=/var/log/nginx/error.log"
+    "--pid-path=/var/log/nginx/nginx.pid"
+    "--http-client-body-temp-path=/var/cache/nginx/client_body"
+    "--http-proxy-temp-path=/var/cache/nginx/proxy"
+    "--http-fastcgi-temp-path=/var/cache/nginx/fastcgi"
+    "--http-uwsgi-temp-path=/var/cache/nginx/uwsgi"
+    "--http-scgi-temp-path=/var/cache/nginx/scgi"
   ] ++ optionals withDebug [
     "--with-debug"
   ] ++ optionals withStream [
@@ -105,7 +113,9 @@ stdenv.mkDerivation {
       preInstall = ''
         export nixStoreDir="$NIX_STORE" nixStoreDirLen="''${#NIX_STORE}"
       '';
-    }) ++ optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
+    }) ++ [
+      ./nix-skip-check-logs-path.patch
+    ] ++ optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
       (fetchpatch {
         url = "https://raw.githubusercontent.com/openwrt/packages/master/net/nginx/patches/102-sizeof_test_fix.patch";
         sha256 = "0i2k30ac8d7inj9l6bl0684kjglam2f68z8lf3xggcc2i5wzhh8a";
diff --git a/pkgs/servers/http/nginx/nix-skip-check-logs-path.patch b/pkgs/servers/http/nginx/nix-skip-check-logs-path.patch
new file mode 100644
index 00000000000..a823660cc32
--- /dev/null
+++ b/pkgs/servers/http/nginx/nix-skip-check-logs-path.patch
@@ -0,0 +1,27 @@
+diff --git a/auto/install b/auto/install
+index d884487..dccc411 100644
+--- a/auto/install
++++ b/auto/install
+@@ -148,12 +148,6 @@ install:  build $NGX_INSTALL_PERL_MODULES
+               || cp conf/nginx.conf '\$(DESTDIR)$NGX_CONF_PATH'
+       cp conf/nginx.conf '\$(DESTDIR)$NGX_CONF_PREFIX/nginx.conf.default'
+
+-      test -d '\$(DESTDIR)`dirname "$NGX_PID_PATH"`' \\
+-              || mkdir -p '\$(DESTDIR)`dirname "$NGX_PID_PATH"`'
+-
+-      test -d '\$(DESTDIR)`dirname "$NGX_HTTP_LOG_PATH"`' \\
+-              || mkdir -p '\$(DESTDIR)`dirname "$NGX_HTTP_LOG_PATH"`'
+-
+       test -d '\$(DESTDIR)$NGX_PREFIX/html' \\
+               || cp -R $NGX_HTML '\$(DESTDIR)$NGX_PREFIX'
+ END
+@@ -161,9 +155,6 @@ END
+
+ if test -n "$NGX_ERROR_LOG_PATH"; then
+     cat << END                                                >> $NGX_MAKEFILE
+-
+-      test -d '\$(DESTDIR)`dirname "$NGX_ERROR_LOG_PATH"`' \\
+-              || mkdir -p '\$(DESTDIR)`dirname "$NGX_ERROR_LOG_PATH"`'
+ END
+
+ fi
```

Reply:
We might want to do a separate PR in nixpkgs changing paths to more standard locations before starting with sandboxing.

Reply:
Also, when doing so, pleas see if we can fix the underlying issue `nix-skip-check-logs-path.patch` is trying to workaround.

Reply:
> Also, when doing so, pleas see if we can fix the underlying issue `nix-skip-check-logs-path.patch` is trying to workaround.

Variant - build without sandox.